### PR TITLE
Support IP addresses with subnets in allowed_hosts

### PR DESCRIPTION
**Pull request by mleiner**
This fixes Issue #46.

allowed_hosts of nrpe.cfg can be set to
- hostname
- ip addresses (v4 and v6)
- ip addresses with subnet (v4 and v6)

The current code uses Stdlib::Host which ultimatively resolves to FQDNs and pure IP addresses, but not subnets. It uses Stdlib::Compat, too.

Changing Stdlib::Host to Variant[Stdlib::Fqdn,Stdlib::IP::Address] solves this by
- Resolve Stdlib::Host to Stdlib::Fqdn,Stdlib::Compat::Ip_address
- In the same step replace Stdlib::Compat::Ip_address to Stdlib::IP::Address

This gets rid of the Stdlib::Compat layer and allows `allowed_hosts` to accept IP addresses with subnets, too.

I have not found any spec tests for allowed_hosts, thus didn't change existing ones. I can try to add those if you like.

I've tested locally with 
```
nrpe::allowed_hosts:
  - "::1"
  - "2001::/96"
  - "127.0.0.1"
  - "192.168.0.0/24"
```

Improvement suggestions welcome

### CODE REVIEW

**Comment by egmont1227** [comment]
@alexjfisher @bastelfreak @blkperl can you have a look and merge this ?

File: manifests/init.pp [hunk 1]
```diff
@@ -100,7 +100,7 @@
 # @param service_name
 #   **Private** You should not need to override this parameter.
 class nrpe (
-  Array[Stdlib::Host]                  $allowed_hosts                   = ['127.0.0.1'],
+  Array[Variant[Stdlib::Fqdn,Stdlib::IP::Address]]                  $allowed_hosts                   = ['127.0.0.1'],
```
Comment (bastelfreak) [inline manifests/init.pp:103]:
```suggestion
  Array[Variant[Stdlib::Fqdn,Stdlib::IP::Address]]  $allowed_hosts                   = ['127.0.0.1'],
```

can you please clean it up a bit like this this and maybe remove a bit of the whitespace so the `=` are vertically aligned?

File: templates/nrpe.cfg.epp [hunk 2]
```diff
@@ -5,7 +5,7 @@
   Stdlib::IP::Address            $server_address,
   String                         $nrpe_user,
   String                         $nrpe_group,
-  Array[Stdlib::Host]            $allowed_hosts,
+  Array[Variant[Stdlib::Fqdn,Stdlib::IP::Address]]            $allowed_hosts,
```
Comment (bastelfreak) [inline templates/nrpe.cfg.epp:8]:
```suggestion
  Array[Variant[Stdlib::Fqdn,Stdlib::IP::Address]] $allowed_hosts,
```

**Comment by bastelfreak** [comment]
thanks for the work! can you please add some tests in https://github.com/voxpupuli/puppet-nrpe/blob/master/spec/classes/nrpe_spec.rb that verify it works with the different ip/cidr/fqdn formats?

File: manifests/init.pp (hunk 1 above)
Comment (mleiner) [reply manifests/init.pp]: This was unintended, this is now fixed.

**Comment by mleiner** [comment]
@bastelfreak I've fixed the syntax and added positive spec tests for allowed_hosts. I skipped negative tests since I expect stdlib to catch type mismatches. Is that correct?

**Comment by bastelfreak** [comment]
looks good to me. can you take a look at the indentation in templates/nrpe.cfg.epp ?

**Comment by mleiner** [comment]
I have fixed the template syntax, too. Sorry for not checking while I already worked on init.pp.

**Comment by bastelfreak** [comment]
thanks!

**Comment by mleiner** [comment]
Are you planning a new release?

**Comment by bastelfreak** [comment]
I would like to, but that's currently blocked by https://github.com/voxpupuli/puppet-nrpe/pull/48

**Pull request merged**